Sources/FoundationEssentials/Calendar/Calendar_Gregorian.swift

```diff
@@ -10,26 +10,108 @@
 
+/// Julian date helper
+/// Julian dates are noon-based. Gregorian dates are midnight-based.
+/// JulianDates        `2451544.5 ..< 2451545.5` (Jan 01 2000, 00:00 - Jan 02 2000, 00:00)
+/// maps to JulianDay  `2451545`                 (Jan 01 2000, 12:00)
+extension Date {
+    static let julianDayAtDateReference: Double = 2_451_910.5 // 2001 Jan 1, midnight, UTC
+
+    var julianDate: Double {
+        timeIntervalSinceReferenceDate / 86400 + Self.julianDayAtDateReference
+    }
+
+    var julianDay: Int {
+        Int(julianDate.rounded())
+    }
+
+    init(julianDay: Int) {
+        self.init(julianDate: Double(julianDay))
+    }
+
+    init(julianDate: Double) {
+        let secondsSinceJulianReference = (julianDate - Self.julianDayAtDateReference) * 86400
+        self.init(timeIntervalSinceReferenceDate: secondsSinceJulianReference)
+    }
+}
 /// This class is a placeholder and work-in-progress to provide an implementation of the Gregorian calendar.
 internal final class _CalendarGregorian: _CalendarProtocol, @unchecked Sendable {
+    
+    let julianCutoverDay: Int// Julian day (noon-based) of cutover
+    let gregorianStartYear: Int
+    let gregorianStartDate: Date
+
+    // Only respects Gregorian identifier
     init(identifier: Calendar.Identifier, timeZone: TimeZone?, locale: Locale?, firstWeekday: Int?, minimumDaysInFirstWeek: Int?, gregorianStartDate: Date?) {
-        fatalError()
+
+        self.timeZone = timeZone ?? .gmt
+        if let gregorianStartDate {
+            self.gregorianStartDate = gregorianStartDate
+            self.julianCutoverDay = gregorianStartDate.julianDay
+            let (y, _, _) = Self.yearMonthDayFromJulianDay(julianCutoverDay, useJulianRef: false)
+            self.gregorianStartYear = y
+        } else {
+            self.gregorianStartYear = 1582
+            self.julianCutoverDay = 2299161
+            self.gregorianStartDate = Date(timeIntervalSince1970: -12219292800) // 1582-10-15T00:00:00Z
+        }
+
+        self.locale = locale
+        self.localeIdentifier = locale?.identifier ?? ""
+        if let firstWeekday, (firstWeekday >= 1 && firstWeekday <= 7) {
+            _firstWeekday = firstWeekday
+        }
+
+        if var minimumDaysInFirstWeek {
+            if minimumDaysInFirstWeek < 1 {
+                minimumDaysInFirstWeek = 1
+            } else if minimumDaysInFirstWeek > 7 {
+                minimumDaysInFirstWeek = 7
+            }
+            _minimumDaysInFirstWeek = minimumDaysInFirstWeek
+        }
     }
     
     var identifier: Calendar.Identifier {
         .gregorian
     }
     
     var locale: Locale?
-    
+
     var localeIdentifier: String
     
     var timeZone: TimeZone
     
-    var firstWeekday: Int
-    
-    var minimumDaysInFirstWeek: Int
-    
+    var _firstWeekday: Int?
+    var firstWeekday: Int {
+        set {
+            if newValue >= 1 && newValue <= 7 {
+                _firstWeekday = newValue
+            }
```

Comment:
Is there any value in a `fatalError` here instead of silently dropping the value?